### PR TITLE
Separate accumulator and result types

### PR DESCRIPTION
**Pull request by jopperm**
Separates accumulator and result types in the block pointer example, and enables two additional testcases:
- f16 += f16 x f16, stored as f16
- f32 += f16 x f16, truncated/stored as f16

For the f16 accumulation test, I had to raise the tolerances and limit the number of elements that are accumulated (here: 3), otherwise I wasn't able to keep the accumulation error in check.

### CODE REVIEW

**Review by Dewei-Wang-sh — APPROVED** [review]
lgtm

File: python/tutorials/09-experimental-block-pointer.py [hunk 1]
```diff
@@ -222,32 +223,47 @@ def matmul(a, b, res_dtype):
...
-    triton_output = matmul(a, b, res_dtype)
-    if dtype.is_floating_point:
-        torch.xpu.set_fp32_math_mode(torch.xpu.utils.FP32MathMode.TF32 if dtype ==
-                                     torch.float32 else torch.xpu.utils.FP32MathMode.FP32)
-        torch_output = torch.matmul(a, b).to(res_dtype)
-    else:
         # torch.matmul clamps values to input dtype; IPEX doesn't support int32 matmul
-        torch_output = torch.matmul(a.to(device='cpu', dtype=res_dtype), b.to(device='cpu',
-                                                                              dtype=res_dtype)).to(device='xpu')
+        torch_output = torch_output = torch.matmul(a.to(device='cpu', dtype=accum_dtype),
```
Comment (whitneywhtsang) [inline python/tutorials/09-experimental-block-pointer.py:253]:
```suggestion
       torch_output = torch.matmul(a.to(device='cpu', dtype=accum_dtype),
```

Comment (jopperm) [reply python/tutorials/09-experimental-block-pointer.py]: 🙈

**Review by victor-eds — APPROVED** [review]

File: test/Conversion/intel/tritongpu_to_llvm_intel_block_ptr.mlir [hunk 2]
```diff
@@ -140,3 +140,27 @@ module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 32
     tt.return
   }
 }
+
+// -----
+
+// COM: Checks the correct lowering of a 16-bit 2D-block-store.
+
+module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 32 : i32, triton_gpu.shared = 0 : i32, "triton_gpu.threads-per-warp" = 16 : i32} {
```
Comment (etiotto) [inline test/Conversion/intel/tritongpu_to_llvm_intel_block_ptr.mlir:148]: Remove `triton_gpu.num-ctas`, I do not think is required.

Comment (etiotto) [reply test/Conversion/intel/tritongpu_to_llvm_intel_block_ptr.mlir:148]: Do we need ` triton_gpu.shared = ` ?

File: test/Conversion/intel/tritongpu_to_llvm_intel_block_ptr.mlir [hunk 3]
```diff
@@ -140,3 +140,27 @@ module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 32
...
+  // CHECK-LABEL: llvm.func spir_kernelcc @matmul_kernel_with_block_pointers_f16accu(
+  // CHECK-SAME:                                                                     [[VAL_0:%.*]]: !llvm.ptr<1>) attributes {triton_gen.intel_reqd_sub_group_size = [16 : i32], triton_gen.max_work_group_size = [512 : i32, 1 : i32, 1 : i32]} {
+  tt.func public @matmul_kernel_with_block_pointers_f16accu(%arg0: !tt.ptr<f16> {tt.divisibility = 16 : i32}) {
```
Comment (etiotto) [inline test/Conversion/intel/tritongpu_to_llvm_intel_block_ptr.mlir:151]: can tt.divisibility be removed ?

**Review by whitneywhtsang — APPROVED** [review]

**Review by etiotto — APPROVED** [review]

**Pull request merged**